### DISCUSSION TITLE
Does zod-form-data support arrays of objects?

### DISCUSSION THREAD

**Discussion by JeffBeltran** [airjp73/rvf#129] [Q&A] [answered]
Hey there, curious if im doing things wrong or trying something that isn't supported. 

Looking at https://www.remix-validated-form.io/arrays-and-nested, i thought this would work

```
import { useActionData } from "@remix-run/react";
import { z } from "zod";
import { zfd } from "zod-form-data";

const schema = zfd.formData({
  todos: zfd.repeatable(
    z.array(z.object({ title: zfd.text(), description: zfd.text() }))
  ),
});

export const action: ActionFunction = async ({ request }) => {
  const formData = await request.formData();
  return schema.parse(formData);
};

export default function Foo() {
  const data = useActionData();

  console.log(data);

  return (
    <form method="post">
      <input name="todos[0].title" defaultValue="title 1" />
      <input name="todos[0].description" defaultValue="description 1" />
      <input name="todos[1].title" defaultValue="title 2" />
      <input name="todos[1].description" defaultValue="description 2" />
      <button type="submit">Submit</button>
    </form>
  );
}
```

when i submit the form i get

```
{ todos: [] }
```

i've tried a few different combos on the schema but can't seem to get what i would expect, so just not sure if i'm using the package wrong? or if i need to use the full remix-validated-form package?

**Comment by airjp73** [comment, marked as answer]
Looks like this is a bug. This line [here](https://github.com/airjp73/remix-validated-form/blob/main/packages/zod-form-data/src/helpers.ts#L170) should be using `set` from lodash. Would you mind filing an issue?

**Comment by tsulatsitamim** [comment]
I still can get empty array when validate array of object

result
![Screenshot 2024-04-25 at 07 06 27](https://github.com/airjp73/remix-validated-form/assets/5946791/ae7aa227-13fe-4952-be2b-3140e9f55bb0)

schema
![Screenshot 2024-04-25 at 07 05 08](https://github.com/airjp73/remix-validated-form/assets/5946791/240b8b0d-b7bf-41fa-8a1d-2ac4f406ace6)

payload
![Screenshot 2024-04-25 at 07 04 54](https://github.com/airjp73/remix-validated-form/assets/5946791/6bdcc591-ad48-4090-b7fc-7d192f2f6016)
